libraries/AP_Follow/AP_Follow.cpp

```diff
@@ -611,19 +619,36 @@ bool AP_Follow::handle_global_position_int_message(const mavlink_message_t &msg)
         return false;
     }
 
+    // apply jitter-corrected timestamp to this update
+    uint32_t location_update_ms = _jitter.correct_offboard_timestamp_msec(packet.time_boot_ms, AP_HAL::millis());
+    if (location_update_ms < _last_location_update_ms) {
+        // ignore if the new update is older than the most recent one received
+        return false;
+    }
+    _last_location_update_ms = location_update_ms;
+
     Location _target_location;
     _target_location.lat = packet.lat;
     _target_location.lng = packet.lon;
 
-    // set target altitude based on configured altitude type
-    if (_alt_type == AP_FOLLOW_ALTITUDE_TYPE_RELATIVE) {
-        // use relative altitude above home
-        _target_location.set_alt_cm(packet.relative_alt / 10, Location::AltFrame::ABOVE_HOME);
-    } else {
-        // use absolute altitude
-        _target_location.set_alt_cm(packet.alt / 10, Location::AltFrame::ABSOLUTE);
+    switch((Location::AltFrame)_alt_type) {
+        case Location::AltFrame::ABSOLUTE:
+            _target_location.set_alt_cm(packet.alt * 0.1, Location::AltFrame::ABSOLUTE);
+            break;
+        case Location::AltFrame::ABOVE_HOME:
+            _target_location.set_alt_cm(packet.relative_alt * 0.1, Location::AltFrame::ABOVE_HOME);
+            break;
+#if APM_BUILD_TYPE(APM_BUILD_ArduPlane|APM_BUILD_ArduCopter)
```

Comment:
Is this a pattern we use elsewhere?  Does this even work?

Usually this would be `||` across two calls to APM_BUILD_TYPE

Reply:
Yes looks like this won't work, is there a way it could throw an error, because it seems like an easy mistake to make.
Anyway, changed it.